ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py

```diff
@@ -275,7 +277,15 @@ def _yield_lineage_from_query(
 
         database_name = database.details.db if database and database.details else None
 
-        lineage_parser = LineageParser(query)
+        db_service = self.metadata.get_by_name(DatabaseService, db_service_name)
+
+        lineage_parser = LineageParser(
+            query,
+            ConnectionTypeDialectMapper.dialect_of(
+                db_service.connection.config.type.value
```

Comment:
```suggestion
                db_service.serviceType.value
```

as `connection` is an optional field but `serviceType` is a required field in the DatabaseService object
and will contain the same value, @LucasGarcia07 can you please validate and test is out once if this works?

Reply:
also the tests are failing can you please check that as well?

Reply:
thanks for the review, i will fix and test!